inst/htmlwidgets/plotly.js

```diff
@@ -300,3 +272,163 @@ HTMLWidgets.widget({
   }
 
 });
+
+
+
+/**
+ * @param graphDiv The Plotly graph div
+ * @param group The crosstalk group object
+ */
+function TraceManager(graphDiv) {
+  // The Plotly graph div
+  this.gd = graphDiv;
+
+  // Preserve the original data. We'll subset based off of this whenever
+  // filtering is (re)applied.
+  this.origData = JSON.parse(JSON.stringify(graphDiv.data));
+
```

Comment:
Yes. It's a bit questionable to do it this way as it loses any functions, Date objects, and any other custom objects that aren't JSON, but it didn't look to me like the `.data` property would have any of that anyway...?